docs/IDENTITY_PROIVIDER.md

```diff
@@ -0,0 +1,184 @@
+           1. Click the `Create credentials` option in top nav bar
+              1. Select `OAuth client ID`
+              2. Select `Web application` type
+              3. Add name of client
+              4. Add authorized redirect URIs ( based on IDP configuration in keycloak )    Example values:
+                 1. OpenID Connect v1.0 = https://sso.uds.dev/realms/uds/broker/{idp name}/endpoint
+                 2. Google Social = https://sso.uds.dev/realms/uds/broker/google/endpoint
+              5. Hit `Save`
+        3. Capture the client ID and client secret for use in setting up IDP ( these can be accessed later )
+
+---
+
+# Configure Keycloak Google SSO
+
+## Configure the Keycloak Realm.json for Automation
+
+Example of using a secret for supplying the configuration values.
+
+1. This section has been added to the realm json
+
+    ```
+    "identityProviders": [
+        {
+          "alias": "oidc",
+          "displayName": "Google SSO",
+          "internalId": "698ce16e-b026-43d5-8c8c-7977a2659a1c",
+          "providerId": "oidc",
+          "enabled": "${REALM_GOOGLE_IDP_ENABLED}",
+          "updateProfileFirstLoginMode": "on",
+          "trustEmail": true,
+          "storeToken": false,
+          "addReadTokenRoleOnCreate": false,
+          "authenticateByDefault": false,
+          "linkOnly": false,
+          "config": {
+            "tokenUrl": "https://oauth2.googleapis.com/token",
+            "acceptsPromptNoneForwardFromClient": "false",
+            "jwksUrl": "https://www.googleapis.com/oauth2/v3/certs",
+            "isAccessTokenJWT": "false",
+            "filteredByClaim": "false",
+            "backchannelSupported": "false",
+            "issuer": "https://accounts.google.com",
+            "loginHint": "false",
+            "clientAuthMethod": "client_secret_post",
+            "syncMode": "IMPORT",
+            "clientSecret": "${REALM_GOOGLE_IDP_CLIENT_SECRET}",
+            "allowedClockSkew": "0",
+            "defaultScope": "openid profile email",
+            "userInfoUrl": "https://openidconnect.googleapis.com/v1/userinfo",
+            "validateSignature": "true",
+            "hideOnLoginPage": "false",
+            "clientId": "${REALM_GOOGLE_IDP_CLIENTID}",
+            "uiLocales": "false",
+            "disableNonce": "false",
+            "useJwksUrl": "true",
+            "sendClientIdOnLogout": "false",
+            "pkceEnabled": "false",
+            "metadataDescriptorUrl": "https://accounts.google.com/.well-known/openid-configuration",
+            "authorizationUrl": "https://accounts.google.com/o/oauth2/v2/auth",
+            "disableUserInfo": "false",
+            "sendIdTokenOnLogout": "true",
+            "passMaxAge": "false"
+          }
+        }
+    ],
+    ```
+    * note the `${REALM_GOOGLE_IDP_ENABLED}`, `${REALM_GOOGLE_IDP_CLIENT_SECRET}`, and `${REALM_GOOGLE_IDP_CLIENTID}` pieces that are provided by uds-core environment variables
+
+2. Update `uds-core` 
```

Comment:
I think this part can be removed, and you should just skip to showing how the `realmInitEnv` is configured via the values.